alacritty_terminal/src/term/search.rs

```diff
@@ -450,41 +468,41 @@ impl<T> Term<T> {
 
         let wide = Flags::WIDE_CHAR | Flags::WIDE_CHAR_SPACER | Flags::LEADING_WIDE_CHAR_SPACER;
         while let Some(cell) = iter.prev() {
-            if !cell.flags.intersects(wide) && self.semantic_escape_chars.contains(cell.c) {
-                break;
-            }
+            point = cell.point;
 
-            if cell.point.column == last_column && !cell.flags.contains(Flags::WRAPLINE) {
-                break; // cut off if on new line or hit escape char
+            if !cell.flags.intersects(wide) && needles.contains(cell.c) {
+                return Ok(point);
             }
 
-            point = cell.point;
+            if point.column == last_column && !cell.flags.contains(Flags::WRAPLINE) {
+                break;
+            }
```

Comment:
This code doesn't match all semantics for the `semantic_search` since it seems like we need to compensate inside the `break` as well, because the point will be different.

Though, the simple fix would be to move the `point = cell.point` like it was before, and do a `return Ok(cell.point)`.

This will be exactly like before when we account for compensation for `Ok` you have.

Reply:
I've intentionally changed it though, otherwise I would have kept it the same? Can you actually provide an example where this introduces a regression? Because my testing didn't show any changes.

Reply:
I haven't found it myself, because only one part about last column is different, but last column means that we reached end anyway. The other code suggests that as well though.